### PR TITLE
Working rigid links with new name/ID convention

### LINKED ISSUE #188: Cannot Pull or Push Rigid Links to SAP2000 v21
#### Description:
It seems that using SAP2000 v21 with BHoM Beta V.4.0.1.0, rigid links elements cannot be Pull or Push from analysis model.

#### Expected behaviour:
Be able to Pull or Push Rigid links elements from analysis model.

### PR DESCRIPTION
**Pull request by maryannewachter**
### Issues addressed by this PR

Closes #188 

### Test files
https://burohappold.sharepoint.com/sites/BHoM/02_Current/Forms/AllItems.aspx?RootFolder=%2Fsites%2FBHoM%2F02%5FCurrent%2F12%5FScripts%2F01%5FIssue%2FBHoM%2FSAP2000%5FToolkit%2F%23188%2DRigidLink%2Dpull%2Dpush%2Derror&FolderCTID=0x0120008122C8891F89054B8ACED0196C70DFC4

### Additional comments

Breaking change was due to changes with AdapterID fragments

### CODE REVIEW

File: SAP2000_Engine/SAP2000_Engine.csproj [hunk 1]
```diff
@@ -95,6 +95,11 @@
       <HintPath>C:\ProgramData\BHoM\Assemblies\Results_Engine.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="SAP2000_oM, Version=4.0.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\..\..\..\ProgramData\BHoM\Assemblies\SAP2000_oM.dll</HintPath>
+      <Private>False</Private>
+    </Reference>
```
Comment (FraserGreenroyd) [inline SAP2000_Engine/SAP2000_Engine.csproj:102]: This should be added as a project reference to avoid potential CI issues down the line :smile:

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Review by JosefTaylor — APPROVED** [review]
Looks great, I simplified the filtering LINQ query at the end- please review!

**Comment by maryannewachter** [comment]
New LINQ query LGTM.

File: SAP2000_Engine/SAP2000_Engine.csproj [hunk 2]
```diff
@@ -95,6 +95,10 @@
       <HintPath>C:\ProgramData\BHoM\Assemblies\Results_Engine.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="SAP2000_oM">
+      <HintPath>C:\ProgramData\BHoM\Assemblies\SAP2000_oM.dll</HintPath>
+      <Private>False</Private>
+    </Reference>
```
Comment (FraserGreenroyd) [inline SAP2000_Engine/SAP2000_Engine.csproj:101]: This shouldn't be necessary at all with the project reference? Does it not work without this?

Comment (JosefTaylor) [reply SAP2000_Engine/SAP2000_Engine.csproj:101]:
```suggestion
```

**Review by JosefTaylor — CHANGES_REQUESTED** [review]
should be able to simply delete

**Review by JosefTaylor — APPROVED** [review]

**Comment by JosefTaylor** [comment]
@BHoMBot check compliance
@BHoMBot check installer

**Comment by JosefTaylor** [comment]
@BHoMBot check core

**Comment by JosefTaylor** [comment]
@BHoMBot check compliance

**Comment by JosefTaylor** [comment]
@BHoMBot check installer

**Review by FraserGreenroyd — APPROVED** [review]
Can we raise an issue to address the failing compliance issue at a future date? If so I'm happy to approve this from a CI perspective with the last changes to the `.csproj` file :smile:

Oh, and a label please for the change log!

**Comment by JosefTaylor** [comment]
@FraserGreenroyd yeah- I looked at the issue and we either need to make these methods private or work out some grasshopper issues. I'd appreciate if you let us ignore that check, and I'll raise an issue!

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a CI/CD instruction. I am authorising dispensation to be granted on check ref. 1705030356

**Comment by JosefTaylor** [comment]
new issue raised for the compliance issue: #190

**Comment by FraserGreenroyd** [comment]
Cheers @JosefTaylor :smile:

**Pull request merged**